src/client/executor.rs

```diff
@@ -637,6 +660,7 @@ impl Error {
             TransactionState::Starting | TransactionState::InProgress => {
                 if self.is_network_error() || self.is_server_selection_error() {
                     self.add_label(TRANSIENT_TRANSACTION_ERROR);
+                    session.as_mut().unwrap().unpin_mongos();
```

Comment:
the error may already have had a `TransientTransactionError` label appended to it via the server, so I think we actually need to move this unpin outside of the match on the transaction state. `UnknownTransactionCommitResult` is only appended by the driver so its technically not necessary to move that one too, but it could be nice to do so for symmetry with the other label.

Reply:
Also noticed this and made the fix, so I just saw this - I think it makes sense to move the other unpin out as well, so I'll go ahead and do that.